src/open-feature.ts

```diff
@@ -0,0 +1,59 @@
+import { OpenFeatureClient } from './client';
+import { NOOP_PROVIDER } from './no-op-provider';
+import { Client, Context, EvaluationLifeCycle, FlagValue, Hook, Provider } from './types';
+
+// use a symbol as a key for the global singleton
+const GLOBAL_OPENFEATURE_API_KEY = Symbol.for('@openfeature/js.api');
+
+type OpenFeatureGlobal = {
+  [GLOBAL_OPENFEATURE_API_KEY]?: OpenFeature;
+};
+const _global = global as OpenFeatureGlobal;
+
+export class OpenFeature implements EvaluationLifeCycle {
+  private _provider: Provider = NOOP_PROVIDER;
+  private _context: Context = {};
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  private constructor() {}
+
+  static get instance(): OpenFeature {
+    const globalApi = _global[GLOBAL_OPENFEATURE_API_KEY];
+    if (globalApi) {
+      return globalApi;
+    }
+
+    const instance = new OpenFeature();
+    _global[GLOBAL_OPENFEATURE_API_KEY] = instance;
+    return instance;
+  }
+
+  getClient(name?: string, version?: string, context?: Context): Client {
+    return new OpenFeatureClient(this, { name, version }, context);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  addHooks(...hooks: Hook<FlagValue>[]): void {
+    throw new Error('Method not implemented.');
+  }
+
+  get hooks(): Hook<FlagValue>[] {
+    throw new Error('Method not implemented.');
+  }
+
+  set provider(provider: Provider) {
+    this._provider = provider;
+  }
+
+  get provider(): Provider {
+    return this._provider;
+  }
+
+  set context(context: Context) {
```

Comment:
Is your thinking that you can set an evaluation context on the api as well? I seemed to remember that we had a cascade of these where you could set the context on the API and client too, but I couldn't find it in the spec.

Reply:
I was looking for it yesterday. :) We can accomplish the same with hooks though it will be maybe a little jankier. 

If/when we add it, there are some things that will need to be updated around hook context merges. No rush on my part for now.